solved_ATM_ex/solved_ATM_ex.py

```diff
@@ -0,0 +1,170 @@
+import os
+import sys
+import json
+
+ATM_FILE_PLACE = 1
+BALANCE_OF_COSTUMER = 'balance'
+PASSWORD_OF_COSTUMER = 'password'
+COSTUMER_CHOICE_OPTIONS = "1 - View balance \n" \
+                          "2 - Withdraw \n" \
+                          "3 - Deposit \n" \
+                          "4 - Change password \n" \
+                          "5 - Exit\n"
+SWITCHER = {
+    '1': lambda data, identifier: print(data[identifier][
+                                            BALANCE_OF_COSTUMER]),
+    '2': lambda data, identifier: cash_withdraw(data, identifier),
+    '3': lambda data, identifier: cash_deposit(data, identifier),
+    '4': lambda data, identifier: change_password(data, identifier)
+}
+
+
+def check_convert_to_float(amount):
+    """Check if variable can be converted to float.
+
+    Args:
+        amount (str): Costumer's input.
+
+    Returns:
+        float: Change the type of the variable 'amount' and return it,
+            else if couldn't- return None.
+    """
+    try:
+        return float(amount)
+
+    except ValueError:
+        return None
+
+
+def cash_withdraw(costumers_data, costumer_id):
+    """Withdraw amount of money from the ATM .
+
+    Args:
+        costumers_data (dict): Bank costumer data.
+        costumer_id (str): Costumer's ID.
+    """
+    amount_to_withdraw = input("Amount to withdraw: ")
+    check_amount = check_convert_to_float(amount_to_withdraw)
+    balance_now = float(costumers_data[costumer_id][BALANCE_OF_COSTUMER])
+    if check_amount and 0 < check_amount <= balance_now:
+        new_balance = balance_now - check_amount
+        costumers_data[costumer_id][BALANCE_OF_COSTUMER] = new_balance
+        update_changes_in_dictionary(costumers_data)
+
+    else:
+        print("Failed - check amount\n")
+
+
+def cash_deposit(costumers_data, costumer_id):
+    """Deposit money to the bank.
+
+    Args:
+        costumers_data (dict): Bank costumer data.
+        costumer_id (str): Costumer's ID.
+    """
+    amount_to_deposit = input("Amount to deposit: ")
+    check_amount = check_convert_to_float(amount_to_deposit)
+    if check_amount and check_amount > 0:
+        balance_now = float(costumers_data[costumer_id][
+                                BALANCE_OF_COSTUMER])
+        new_balance = balance_now + check_amount
+        costumers_data[costumer_id][BALANCE_OF_COSTUMER] = new_balance
+        update_changes_in_dictionary(costumers_data)
+
+    else:
+        print("Failed - check amount\n")
+
+
+def change_password(costumers_data, costumer_id):
+    """Change costumer's password.
+
+    Args:
+        costumers_data (dict): Bank costumer data.
+        costumer_id (str): Costumer's ID.
+    """
+    new_password = input("New password: ")
+    if len(new_password) == 4 and new_password.isdigit():
+        costumers_data[costumer_id][PASSWORD_OF_COSTUMER] = new_password
```

Comment:
This is a source for bugs.
Its correct that if you change dict in function you will change the source dict, but it make it harder to miss it if there is a bug.
If you got function you want to update a dict inside of it, you should return the updated dict at the end, this way its more clear where you change it, and easier to debug if there is a problem